### PR TITLE
feat(AUTH-CPC-731): auth reset password

### PR DESCRIPTION
**Pull request by DylanBrass**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/CPC-731?atlOrigin=eyJpIjoiNzc0OTc2YzNhNGFkNDk5YTg0YzVhMTExMmY0ZDI4YmIiLCJwIjoiaiJ9)
## Context:
The user can now reset their password. This is done with an email with a link with a token that will validate the user. The token is hashed and stored in the database. The password is then updated.

## Changes

- Created new table for tokens
- Made all requests in api-gateway reactive for auth-service
- Made the UI for the reset password process
- Send email to user
- Made user with my email to test the forgot password (please don't abuse, it will be deleted later)

#### Before : 
image
#### After : 
image
image
image
image

### CODE REVIEW

File: api-gateway/src/main/resources/static/images/icons8-trash-32.png
Comment (nic5694) [inline api-gateway/src/main/resources/static/images/icons8-trash-32.png:1]: i would suggest maybe instead using a SVG i think that its a icon that your trying to use but to make it as light as possible i would suggest an SVG. this format is better and lighter and used more in front end development

File: api-gateway/src/main/java/com/petclinic/bffapigateway/utils/Utility.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.petclinic.bffapigateway.utils;
+
+
+import org.springframework.http.server.reactive.ServerHttpRequest;
+
+public class Utility {
+    public static String getSiteURL(ServerHttpRequest request) {
+        return request.getURI().toString();
```
Comment (cubajosh) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/utils/Utility.java:8]: can you send me a message explaining what this does?

**Review by cubajosh — APPROVED** [review]
very well done man, everything looks good but some stuff I don't entirely understand. Can you explain some of this to me when you have time?

**Review by AmirHamzahMirza — APPROVED** [review]
Looks good and with the passing tests it shouldn't prove to have any issues

File: api-gateway/src/main/resources/static/scripts/auth/forgot-pwd-form/forgot-pwd-form.controller.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+angular.module('forgotPwdForm')
+    .controller('forgotPwdFormController', ["$http", '$location', "$scope", function ($http, $location, $scope) {
```
Comment (nic5694) [inline api-gateway/src/main/resources/static/scripts/auth/forgot-pwd-form/forgot-pwd-form.controller.js:4]: usually ive seen in the project the controller names to be all capital but im not sure

File: auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java [hunk 3]
```diff
@@ -202,9 +205,197 @@ public HashMap<String,Object> login(UserIDLessUsernameLessDTO login) throws Inco
...
+                             </ol>
+                            \s
+                             <p>If you did not request this password reset, please disregard this email. Your account security is important to us, and no changes will be made without your verification.</p>
+                            \s
+                             <p>Thank you for choosing Pet Clinic.</p>
+                         </div>
+                     </body>
+                     </html>
+                     """, link);
+
```
Comment (nic5694) [inline auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java:375]: is the email responsive im just thinking for the people who open this on their phone can they read it and make it work?

Comment (nic5694) [reply auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java:375]: this is a detail i think if its not you should add it to your sprint backlog

**Review by nic5694 — APPROVED** [review]
take a look at my comments, most of them are details and could be added to your sprint backlog if you like the ideas i brought

File: api-gateway/src/main/java/com/petclinic/bffapigateway/utils/Utility.java (hunk 1 above)
Comment (DylanBrass) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/utils/Utility.java:8]: With the ServerHttpRequest you can get the the URL, but in this case I don't need it haha. This was used before to send the link in the email, but I had to remove it.

File: auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java (hunk 3 above)
Comment (DylanBrass) [reply auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java:375]: It is ! There is little to no styling, but I am thinking of making it better.

File: api-gateway/src/main/resources/static/images/icons8-trash-32.png
Comment (DylanBrass) [reply api-gateway/src/main/resources/static/images/icons8-trash-32.png:1]: That wasn't done by me lol, a quirk of git.

File: auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java (hunk 3 above)
Comment (nic5694) [reply auth-service/src/main/java/com/auth/authservice/businesslayer/UserServiceImpl.java:375]: thats good excited to hear more

**Pull request merged**